components/brave_wallet/browser/swap_response_parser.cc

```diff
@@ -60,77 +123,455 @@ bool ParseSwapResponse(const std::string& json,
+  if (!sell_amount)
+    return nullptr;
+  swap_response->sell_amount = *sell_amount;
+
+  auto allowance_target = ParseResultFromDict(response_dict, "allowanceTarget");
+  if (!allowance_target)
+    return nullptr;
+  swap_response->allowance_target = *allowance_target;
+
+  auto sell_token_to_eth_rate =
+      ParseResultFromDict(response_dict, "sellTokenToEthRate");
+  if (!sell_token_to_eth_rate)
+    return nullptr;
+  swap_response->sell_token_to_eth_rate = *sell_token_to_eth_rate;
+
+  auto buy_token_to_eth_rate =
+      ParseResultFromDict(response_dict, "buyTokenToEthRate");
+  if (!buy_token_to_eth_rate)
+    return nullptr;
+  swap_response->buy_token_to_eth_rate = *buy_token_to_eth_rate;
+
+  return swap_response;
+}
+
+mojom::JupiterSwapQuotePtr ParseJupiterSwapQuote(const std::string& json) {
+  //  {
+  //    "data": [
+  //      {
+  //        "inAmount": 10000,
+  //        "outAmount": 261273,
+  //        "amount": 10000,
+  //        "otherAmountThreshold": 258660,
+  //        "outAmountWithSlippage": 258660,
+  //        "swapMode": "ExactIn",
+  //        "priceImpactPct": 0.008955716118219659,
+  //        "marketInfos": [
+  //          {
+  //            "id": "2yNwARmTmc3NzYMETCZQjAE5GGCPgviH6hiBsxaeikTK",
+  //            "label": "Orca",
+  //            "inputMint": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
+  //            "outputMint": "MNDEFzGvMt87ueuHvVU9VcTqsAP5b3fTGPsHuuPA5ey",
+  //            "notEnoughLiquidity": false,
+  //            "inAmount": 10000,
+  //            "outAmount": 117001203,
+  //            "priceImpactPct": 1.196568750220778e-7,
+  //            "lpFee": {
+  //              "amount": 30,
+  //              "mint": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
+  //              "pct": 0.003
+  //            },
+  //            "platformFee": {
+  //              "amount": 0,
+  //              "mint": "MNDEFzGvMt87ueuHvVU9VcTqsAP5b3fTGPsHuuPA5ey",
+  //              "pct": 0
+  //            }
+  //          }
+  //        ]
+  //      }
+  //    ],
+  //    "timeTaken": 0.044471802000089156
+  //  }
+
+  // STEP 1: convert JSON integer fields in routes to string values.
+  auto converted_json = ConvertMultiUint64InObjectArrayToString(
+      "/data", {"inAmount", "outAmount", "amount", "otherAmountThreshold"},
+      json);
+  if (!converted_json) {
+    LOG(ERROR) << "Failed to parse integer fields, JSON is: " << json;
+    return nullptr;
+  }
+
+  // STEP 2: Obtain number of routes in swap quote.
+  auto routes_value_1 = GetRoutesFromJupiterSwapQuote(*converted_json);
+  if (!routes_value_1)
+    return nullptr;
 
-    if (!ParseResultFromDict(response_dict, "to", &response->to))
-      return false;
+  // STEP 3: Loop over the routes and convert JSON integer fields in
+  // marketInfos to string values.
+  for (int i = 0; i < static_cast<int>(routes_value_1->size()); i++) {
+    converted_json = ConvertMultiUint64InObjectArrayToString(
```

Comment:
This is converting the request payload, not the response. Unless I'm misunderstanding something, I don't think the conversion callback method works here with `api_request_helper_.Request`. We could extract out the conversion logic out of this function - would that be sufficient?

Reply:
I'm confused, this step was part of response parsing and you have moved it to ConvertJupiterQuoteUint64ToString as asked.

Reply:
There used to be two functions in this file:
- `ParseJupiterQuote` - was decoding response body of `GET /v1/quote`
- `EncodeJupiterTransactionParams` - was encoding request payload to `POST `/v1/swap`.

I then moved `EncodeJupiterTransactionParams` to `swap_request_helper.cc` as requested in a review comment, since it was not parsing response. See commit 255b1079111c0305c7168ad7c28ee2bfd502a5c8.

The `ConvertJupiterQuoteUint64ToString` function is not used by the `POST /v1/swap` endpoint. I believe you're confusing it with the work I did in a different commit (616aa95aba9c8f6b1dcac5fba9446ccf21dea466) for `GET /v1/quote`.

Reply:
My DITTO comment was made for the STEP3 in ParseJupiterQuote, not the Encode*, I would blame github if it confuses you.